tests/lsp/test_diagnostics.py

```diff
@@ -0,0 +1,65 @@
+import shutil
+import tempfile
+from pathlib import Path
+
+
+from sqlmesh import Context
+from sqlmesh.core.linter.helpers import read_range_from_file
+from sqlmesh.lsp.context import LSPContext
+from sqlmesh.lsp.uri import URI
+
+
+def test_diagnostic_on_sushi() -> None:
+    # Copy sushi example to a temporary directory
+    sushi_original_path = Path(__file__).parent.parent.parent / "examples" / "sushi"
+
+    tempfolder = tempfile.TemporaryDirectory()
+    temp_path = Path(tempfolder.name)
```

Comment:
FYI, there is a `tmp_path` pytest fixture you can use here. May help cleaning up the testing setup by avoiding the `tempfile` and `copytree` gymnastics.